runtime/src/main/scala/akka/grpc/internal/Gzip.scala

```diff
@@ -15,9 +15,8 @@ object Gzip extends Codec {
   override def compress(uncompressed: ByteString): ByteString = {
     val baos = new ByteArrayOutputStream(uncompressed.size)
     val gzos = new GZIPOutputStream(baos)
-    gzos.write(uncompressed.toArray)
-    gzos.flush()
-    gzos.close()
+    try gzos.write(uncompressed.toArray)
+    finally gzos.close()
```

Comment:
:+1: Good practice and also necessary here to free buffer from native gzip implementation bits immediately (which would only be freed by finalization otherwise).

Reply:
I'll open a separate PR for this

Reply:
changed my mind, just changed the title here

Reply:
Good to close also in case of exceptions, and good catch that flushing indeed doesn't seem necessary (https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/zip/DeflaterOutputStream.html#close() ) :+1: